src/components/dock/dock.tsx

```diff
@@ -0,0 +1,186 @@
+    public host: HTMLLimelDockElement;
+
+    /**
+     * Items that are placed in the dock.
+     */
+    @Prop()
+    public dockItems: DockItem[] = [];
+
+    /**
+     * Items that are placed at the bottom of the dock. (Or at the end in mobile
+     * layout.)
+     */
+    @Prop()
+    public footerItems: DockItem[] = [];
+
+    /**
+     * Defines the width of the component, when it loads.
+     * - `true`: shows both icons and labels of the Dock items.
+     * - `false`: only shows icons of the doc items, and displays
+     * their labels as tooltip.
+     *
+     * Note: when `useMobileLayout` is `true`, labels will always
+     * be shown as tooltips. Read more below…
+     */
+    @Prop({ reflect: true })
+    public expanded = false;
+
+    /**
+     * Set to `false` if you do not want to allow end-users
+     * to exapnd or shrink the Dock. This will hide the
+     * expand/shrink button, and the only things that defines
+     * the layout will be the `expanded` property, and
+     * the `mobileBreakPoint`.
+     */
+    @Prop({ reflect: true })
+    public allowResize = true;
+
+    /**
+     * Defines the breakpoint in pixles, at which the component will be rendered
+     * in a hoizontal layout. Default breakpoint is `700` pixels, which means
+     * when the screen size is smaller than `700px`, the component will automatically
+     * switch to a horizontal layout.
+     */
+    @Prop({ reflect: true })
+    public mobileBreakPoint: number = DEFAULT_MOBILE_BREAKPOINT;
+
+    /**
+     * A document may have several <nav> elements, for example,
+     * one for site navigation and one for intra-page navigation.
+     * `ariaLabelledBy` can be used in such case to promote accessibility,
+     */
+    @Prop({ reflect: true })
+    public ariaLabelledBy: string;
+
+    /**
+     * Fired when a Dock item has been selected from the dock.
+     */
+    @Event()
+    private selected: EventEmitter<DockItem>;
+
+    /**
+     * Is used to render the component horizontally, and place
+     * the Dock items in a row.
+     */
+    @State()
+    private useMobileLayout = false;
+
+    /**
+     * Used to trigger mobileLayout, when viewport width is changed.
+     */
+    private observer: ResizeObserver;
+
+    public componentDidLoad() {
+        this.observer = new ResizeObserver(this.handleResize);
+        this.observer.observe(document.body);
+    }
+
+    disconnectedCallback() {
+        this.observer.disconnect();
+    }
+
+    public render() {
+        return (
+            <Host
+                class={{
+                    dock: true,
+                    expanded: this.expanded,
+                    'has-mobile-layout': this.useMobileLayout,
+                }}
+            >
+                <nav aria-labelledby={this.ariaLabelledBy}>
+                    {this.dockItems.map(this.renderDockItem)}
+                    <span class="footer-separator" />
+                    {this.footerItems.map(this.renderDockItem)}
```

Comment:
The `<span class="footer-separator" />` is hidden, but is used to target the first item that comes after it, applying the extra margin to it.